js/plugins/googleai/src/gemini.ts

```diff
@@ -67,6 +68,10 @@ import { getApiKeyFromEnvVar } from './common';
 import { handleCacheIfNeeded } from './context-caching';
 import { extractCacheConfig } from './context-caching/utils';
 
+type ExtendedUsageMetadata = UsageMetadata & {
```

Comment:
Why does this need to be a new type? What happens when we have another field, is it then ExtendedExtendedUsageMetadata? :)